### PR TITLE
Comments block tweaks

### LINKED ISSUE #7423: Fullpost: Comment Tweaks
Possible to implement these tweaks to comment display?

Design:
comments

Current Site:
screen shot 2016-08-11 at 3 11 50 pm
Example: https://calypso.live/read/feeds/2928910/posts/1117437071#comments

Changes:

• Add header
If 1: "1 COMMENT"
If more than one: "X COMMENTS"
If 0, then "NO COMMENTS – Add the first!"
Standard refresh header CSS: SF UI Text, Medium, 14px, Gray Lighten 10 (#a8bece)

• Tweak to author display
Use same link colors as the post author.
If the commenter has a URL, pull in the site title. (Possible?)
Datestamp on 2nd line, 13px, Gray Lighten 20

• If the commenter has a URL, the gravatar should be linked, too.

• Remove extra space above Reply and Like.

• Remove "open in new window" icon.

• I'd like the links on the commenter's name and site to open their site in Reader, if we're able. Is that possible?

### PR DESCRIPTION
**Pull request by bluefuton**
As per @fraying's issue #7423, this PR will make comments delightful in the following ways:

- [x] Add header showing total number of comments
- [x] Fix author link colours
- ~~Show site title where available (have to check if we have this information)~~ for a later PR
- [x] If the commenter has a URL, the gravatar should be linked, too.
- [x] Remove extra space above Reply and Like
- [x] Remove "open in new window" icon
- [x] Links on the commenter's name and site to open their feed stream in Reader
- [x] Drop date and time to new line

Fixes #7423.

Test live: https://calypso.live/?branch=update/blocks/comments-layout

### CODE REVIEW

**Comment by bluefuton** [comment]
Adding the site information might be one for another PR. We don't currently appear to receive any commenter site information from the API.

**Comment by fraying** [comment]
That’s cool. Listing the site is icing.

**Comment by bluefuton** [comment]
@fraying  when we have a lot of comments to show, we show a link to load earlier comments. Would you like to do any tweaking here? This is what I've got at the moment:

screen shot 2016-08-24 at 17 36 13

For example: http://calypso.localhost:3000/read/feeds/40474296/posts/999143144

**Comment by bluefuton** [comment]
Does the 'no comments' default feel a bit shouty?

screen shot 2016-08-24 at 17 57 23

File: client/blocks/comments/style.scss [hunk 1]
```diff
@@ -350,3 +354,12 @@
 		transform: rotate( 180deg );
 	}
 }
+
+.comments__comment-count {
+	color: lighten( $gray, 10% );
+	size: 13px;
```
Comment (jancavan) [inline client/blocks/comments/style.scss]: Is this supposed to be `font-size`? I think `13px` looks much better though. The original `17px` is a bit much in my opinion.

**Comment by jancavan** [comment]
> Does the 'no comments' default feel a bit shouty?

I think mainly also because it looks large. "A" in add should also be capitalized.

screenshot 2016-08-24 09 51 43

**Comment by fraying** [comment]
Good catch on the "A", @jancavan!

It feels a bit shouty because the other allcaps headers aren't in place yet. Once they're in, it'll blend right in.

**Comment by fraying** [comment]
@bluefuton Thanks for the ping on the "View earlier comments" link. I think it's fine for now.

**Comment by jancavan** [comment]
For Author names, links and non-links are all the same blue. It's not obvious which ones are links. I think for non-links, `darken( $gray, 30% )` works better.

**Comment by fraying** [comment]
Good call, @jancavan. 👍

File: client/blocks/comments/style.scss (hunk 1 above)
Comment (bluefuton) [inline client/blocks/comments/style.scss]: Ah, good catch :) I thought it looked a bit big for 13px...

**Comment by bluefuton** [comment]
Thanks both! I'll follow up on your feedback.

> "A" in add should also be capitalized.

Wearing my pedantic hat, I'd say we should either replace the dash or have a lower-case "a". The first word following a dash isn't usually capitalized:

http://www.quickanddirtytips.com/education/grammar/dashes-colons-and-commas

**Comment by fraying** [comment]
It's true. In this case, the second part ("Add the first!") should really on its own line, below the "NO COMMENTS" header. That doable? If so, it should have the same font settings as the text in a comment.

Or we can leave it like it is.

**Comment by jancavan** [comment]
> The first word following a dash isn't usually capitalized:

Ah, I thought there were exceptions, but TIL 😄

**Comment by bluefuton** [comment]
I reckon 'Add the first!' on a new line is a good idea, but not sure about:

>the same font settings as the text in a comment

...because I believe across Calypso we make the distinction between user-generated content (always in a serif) and interface text (always in a sans-serif).

**Comment by bluefuton** [comment]
All feedback addressed apart from 'add the first'. Thanks both!

File: client/blocks/comments/style.scss (hunk 1 above)
Comment (bluefuton) [inline client/blocks/comments/style.scss]: I've set it to 14px as that's what @fraying originally specced in #7423. That's also the same size as the 'view earlier comments' link.

**Comment by fraying** [comment]
Honestly I'm not worried about that. It's clear from the context that "Add the first!" is not a user-posted comment. And I'm using SF UI Text for some things that are user voice in the refresh. 

Anyway, I'm happy to leave "Add the first!" in the header, but it should have a cap A because it's not a normal sentence, it's a header plus a sentence.

**Comment by fraying** [comment]
@bluefuton Also, the links on the commenters are still opening in a new tab. They should open in the same tab, now that we've removed the "open in new" gridicon. Sorry I didn't make this clear originally.

**Comment by bluefuton** [comment]
> Also, the links on the commenters are still opening in a new tab. They should open in the same tab, now that we've removed the "open in new" gridicon. 

Fixed 👍

**Comment by fraying** [comment]
Thanks! This all LGTM!

**Pull request merged**